src/main/java/core/basesyntax/suplier/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax.suplier;
+
+import core.basesyntax.Figures;
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.Figure;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+
+public class FigureSupplier extends Supplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+        switch (Figures.values()[getRandom().nextInt(Figures.values().length)]) {
+            case CIRCLE:
+                int radius = getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE;
+                figure = new Circle(colorSupplier.getRandomColor(), radius);
+                break;
+            case SQUARE:
+                int side = getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE;
+                figure = new Square(colorSupplier.getRandomColor(), side);
+                break;
+            case RECTANGLE:
+                int width = getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE;
+                int length = getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE;
+                figure = new Rectangle(colorSupplier.getRandomColor(), width, length);
+                break;
+            case RIGHT_TRIANGLE:
+                int firstLeg = getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE;
```

Comment:
maybe we can move `getRandom().nextInt(Supplier.MAX_VALUE) + Supplier.MIN_VALUE` into a separate private method here or public method in parent class?